client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx

```diff
@@ -52,14 +57,14 @@ export const StockItemSearchInput: FC<StockItemSearchInputProps> = ({
   const options = useMemo(
     () =>
       defaultOptionMapper(
-        extraFilter ? items.filter(extraFilter) ?? [] : items ?? [],
+        extraFilter ? (items.filter(extraFilter) ?? []) : (items ?? []),
         'name'
       ).sort((a, b) => a.label.localeCompare(b.label)),
     [items]
   );
 
   const cachedSearchedItems = useMemo(() => {
-    const newItems = [...items, ...(data?.nodes ?? [])];
+    const newItems = data?.nodes ?? [];
```

Comment:
This caused problems whenever we switched the Category, as it didn't remove the old items, just added the new ones. I *think* this change won't cause any regressions elsewhere -- can you think of any?

Reply:
I think this will break the paginated item search, was just looking at this with Aimee so will double check after lunch 🙏

Reply:
It does break pagination (would only affect if you have more than 500 items, but you can test by changing this line:

https://github.com/msupply-foundation/open-msupply/blob/8122021324a96625413720242c02209e07ca1411/client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx#L35

to be:

```ts
  const { pagination, onPageChange } = usePagination(20);
```

You'll be able to scroll down, but then not scroll back up again!

However, I just tried reverting the change, and it seems to be invalidating the list fine? Could you try removing on your end and see if you still see the issue or if something else resolved it?